walrus-attributes-indexer/src/handlers/walrus_blob.rs

```diff
@@ -0,0 +1,471 @@
+
+            let Some(version) = change.input_version else {
+                continue;
+            };
+
+            // This object was previously modified, created, or unwrapped in the checkpoint, so
+            // this version is not a checkpoint input.
+            if output_objects_seen.contains(&id) {
+                continue;
+            }
+
+            // Make sure this object has not already been recorded as an input.
+            let Entry::Vacant(entry) = checkpoint_input_objects.entry(id) else {
+                continue;
+            };
+
+            let input_obj = input_objects_map
+                .get(&(id, version))
+                .copied()
+                .with_context(|| format!(
+                    "Object {id} at version {version} referenced in effects not found in input_objects"
+                ))?;
+
+            entry.insert(input_obj);
+        }
+
+        for change in tx.effects.object_changes() {
+            if change.output_version.is_some() {
+                output_objects_seen.insert(change.id);
+            }
+        }
+    }
+    Ok(checkpoint_input_objects)
+}
+
+/// Extract the file path from the object if it is a walrus metadata dynamic field, otherwise return
+/// None.
+pub fn extract_file_path_and_parent_id(object: &Object) -> Option<(String, ObjectID)> {
+    let Some((metadata, parent_id)) = get_metadata(object).ok()? else {
+        return None;
+    };
+
+    let Some(file_path) = metadata
+        .metadata
+        .get(&"path".to_string())
+        .map(|s| s.to_string())
+    else {
+        return None;
+    };
+
+    Some((file_path, parent_id))
+}
+
+/// Try to deserialize the object as a Walrus Metadata dynamic field, and return the deserialized
+/// data and the parent object ID, or return None if it is not.
+pub fn get_metadata(object: &Object) -> anyhow::Result<Option<(BlobAttribute, ObjectID)>> {
+    let Some(type_) = object.type_() else {
+        return Ok(None);
+    };
+
+    if let Owner::ObjectOwner(parent_id) = object.owner() {
+        // The expected type of the dynamic field is a `Field<DynamicFieldName, BlobAttribute>`.
+        let expected_obj_type = parse_sui_struct_tag(
+            "0x2::dynamic_field::Field<vector<u8>, 0xfdc88f7d7cf30afab2f82e8380d11ee8f70efb90e863d1de8616fae1bb09ea77::metadata::Metadata>"
+        )?;
+
+        if type_.is(&expected_obj_type) {
+            let move_object = object
+                .data
+                .try_as_move()
+                .ok_or_else(|| anyhow::anyhow!("Not a Move object"))?;
+
+            let field: Field<DynamicFieldName, BlobAttribute> =
+                match bcs::from_bytes(move_object.contents()) {
+                    Ok(field) => field,
+                    Err(e) => {
+                        tracing::error!("Failed to deserialize: {:?}", e);
+                        return Err(anyhow::anyhow!("Failed to deserialize: {}", e));
+                    }
+                };
```

Comment:
This is called during `process`, which means the indexing framework will trace the error for you, so you can simplify this to


```suggestion
                bcs::from_bytes(move_object.contents())
                    .context("Failed to deserialize")?;
```

Reply:
neat, thanks for the suggestion